src/sources/MVT.js

```diff
@@ -135,184 +141,36 @@ export default class MVT extends Base {
-        const rs = rsys.getRsysFromTile(x, y, z);
-        const dataframe = this._generateDataFrame(rs, geometries, properties, numFeatures, this._metadata.geomType);
-
-        return dataframe;
-    }
-
-    _decodeMVTLayer (mvtLayer, metadata, mvtExtent) {
-        if (!mvtLayer.length) {
-            return { properties: [], geometries: {}, numFeatures: 0 };
-        }
-        if (!metadata.geomType) {
-            metadata.geomType = this._autoDiscoverType(mvtLayer);
-        }
-        switch (metadata.geomType) {
-            case geometryTypes.POINT:
-                return this._decode(mvtLayer, metadata, mvtExtent, new Float32Array(mvtLayer.length * 2 * 3));
-            case geometryTypes.LINE:
-                return this._decode(mvtLayer, metadata, mvtExtent, [], decodeLines);
-            case geometryTypes.POLYGON:
-                return this._decode(mvtLayer, metadata, mvtExtent, [], decodePolygons);
-            default:
-                throw new Error('MVT: invalid geometry type');
-        }
-    }
-
-    _autoDiscoverType (mvtLayer) {
-        const type = mvtLayer.feature(0).type;
-        switch (type) {
-            case mvtDecoderGeomTypes.point:
-                return geometryTypes.POINT;
-            case mvtDecoderGeomTypes.line:
-                return geometryTypes.LINE;
-            case mvtDecoderGeomTypes.polygon:
-                return geometryTypes.POLYGON;
-            default:
-                throw new Error('MVT: invalid geometry type');
-        }
-    }
-
-    _decode (mvtLayer, metadata, mvtExtent, geometries, decodeFn) {
-        let numFeatures = 0;
-        const { properties, propertyNames } = this._initializePropertyArrays(metadata, mvtLayer.length);
-        for (let i = 0; i < mvtLayer.length; i++) {
-            const f = mvtLayer.feature(i);
-            this._checkType(f, metadata.geomType);
-            const geom = f.loadGeometry();
-            if (decodeFn) {
-                const decodedPolygons = decodeFn(geom, mvtExtent);
-                geometries.push(decodedPolygons);
-            } else {
-                const x = 2 * (geom[0][0].x) / mvtExtent - 1.0;
-                const y = 2 * (1.0 - (geom[0][0].y) / mvtExtent) - 1.0;
-                // Tiles may contain points in the border;
-                // we'll avoid here duplicated points between tiles by excluding the 1-edge
-                if (x < -1 || x >= 1 || y < -1 || y >= 1) {
-                    continue;
+        return this._tileClient.requestData(zoom, viewport, (x, y, z, url) => {
+            return new Promise(resolve => {
+                // Relative URLs don't work inside the Web Worker
+                if (url[0] === '.') {
+                    let parts = window.location.pathname.split('/');
+                    parts.pop();
+                    const path = parts.join('/');
+                    url = window.location.protocol + '//' + window.location.host + '/' + path + '/' + url;
```

Comment:
Having ES6 we can use template literals, like:
``` 
url = `${window.location.protocol}//${window.location.host}/${path}/${url}`
```

Reply:
:+1: